modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/BaselineAutoAdjustTest.java

```diff
@@ -585,6 +586,39 @@ public void shouldJoinFailedBecauseCoordinatorIsInMemoryNodeAndEnabledAutoAdjust
         }
     }
 
+    /**
+     * Test that node joins baseline topology after enabling auto adjust.
+     *
+     * Description:
+     * Start first node and set baseline auto adjust timeout to 100ms. Disable auto adjust and activate cluster.
+     * Start another node and wait until it joins cluster. Enable auto adjust and check that node is in
+     * baseline topology.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testJoinBltExistingNode() throws Exception {
+        IgniteEx ignite0 = startGrid(0);
+
+        ignite0.cluster().baselineAutoAdjustTimeout(100);
+
+        ignite0.cluster().baselineAutoAdjustEnabled(false);
+
+        ignite0.cluster().active(true);
+
+        startGrid(1);
+
+        assertTrue(GridTestUtils.waitForCondition(() -> {
```

Comment:
You can add awaitPartitionMapExchange() after startGrid(1); then you can check ignite0.cluster().nodes().size() without waitForCondition. Also you must check  that currentBaselineTopology().size() equal to 1 before auto-adjust enabled.